datasets/imppres/imppres.py

```diff
@@ -0,0 +1,318 @@
+                    "presupposition": datasets.Value("string"),
+                    "gold_label": datasets.Value("string"),
+                    "UID": datasets.Value("string"),
+                    "pairID": datasets.Value("string"),
+                    "paradigmID": datasets.Value("int16")
+                    # These are the features of your dataset like images, labels ...
+                }
+            )
+        else:  # This is an example to show how to have different features for "first_domain" and "second_domain"
+            features = datasets.Features(
+                {
+                    "premise": datasets.Value("string"),
+                    "hypothesis": datasets.Value("string"),
+                    "gold_label_log": datasets.Value("string"),
+                    "gold_label_prag": datasets.Value("string"),
+                    "spec_relation": datasets.Value("string"),
+                    "item_type": datasets.Value("string"),
+                    "trigger": datasets.Value("string"),
+                    "lexemes": datasets.Value("string"),
+                    # These are the features of your dataset like images, labels ...
+                }
+            )
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        # TODO: This method is tasked with downloading/extracting the data and defining the splits depending on the configuration
+        # If several configurations are possible (listed in BUILDER_CONFIGS), the configuration selected by the user is in self.config.name
+
+        # dl_manager is a datasets.download.DownloadManager that can be used to download and extract URLs
+        # It can accept any type or nested list/dict and will give back the same structure with the url replaced with path to local files.
+        # By default the archives will be extracted and a path to a cached folder where they are extracted is returned instead of the archive
+        # my_urls = _URLs[self.config.name]
+        my_urls = _URLs["default"]
+        data_dir = dl_manager.download_and_extract(my_urls) + "/IMPPRES/" + self.config.name + "/"
```

Comment:
Could you use os.path.join instead of concatenating paths using "/" ?

Reply:
Done!